app/src/main/java/de/westnordost/streetcomplete/MainActivity.java

```diff
@@ -247,6 +253,34 @@ public void onServiceDisconnected(ComponentName className)
 		{
 			questController.deleteOld();
 		}
+
+		Intent intent = getIntent();
+		Uri data = intent.getData();
+		if (Intent.ACTION_VIEW.equals(intent.getAction()) && data != null)
+		{
+			if ("geo".equals(data.getScheme()))
+			{
+				String geoUriRegex = "geo:(-?[0-9]*\\.?[0-9]+),(-?[0-9]*\\.?[0-9]+).*?(?:\\?z=([0-9]*\\.?[0-9]+))?";
```

Comment:
```suggestion
				String geoUriRegex = "^geo:(-?[0-9]*\\.?[0-9]+),(-?[0-9]*\\.?[0-9]+).*?(?:\\?z=([0-9]*\\.?[0-9]+))?";
```
Ensures that `matcher.find()` will only ever find one match because `^` matches to the start of the string.

Reply:
In https://github.com/westnordost/StreetComplete/pull/1396/commits/149eec72d4ba631b1ae24edefcff39a5e4e8f420 I planned to replace `matcher.find()` with `matcher.matches()` because this makes more sense for our use case. But I only removed the `^` at the beginning of the regex (which is not needed when using [`matcher.matches()`](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html#matches--)) and did not replace the method call...

Reply:
Okay, but, for your information:

`matches` matches the whole string against the RegEx, while `find` only searches if a part of the input string matches the RegEx.

The Geo URI scheme is standardized in RFC 5870, see [wikipedia](https://en.wikipedia.org/wiki/Geo_URI_scheme), a URI could look like this:
`geo:37.786971,-122.399677;u=35`.

The `?z=` zoom is not part of the standard so I can't say where it should be expected within the Geo URI. Your RegEx expects it to be at the utter end of the Geo URI, so `geo:37.786971,-122.399677?z=15;u=35` wouldn't match your RegEx.

But I think this is fine because all the examples on the Internet assume those URL query-type parameters like zoom and query to be at the end.